src/_pytest/python.py

```diff
@@ -1101,9 +1106,8 @@ def addcall(self, funcargs=None, id=NOTSET, param=NOTSET):
             invocation through the ``request.param`` attribute.
         """
         if self.config:
-            self.config.warn(
-                "C1", message=deprecated.METAFUNC_ADD_CALL, fslocation=None
-            )
+            self.definition.warn(deprecated.METAFUNC_ADD_CALL)
```

Comment:
this should use stacklevel to show in the hook - the item is the wrong place

Reply:
Good catch, done